src/Microsoft.ML.Data/Transforms/ValueMapping.cs

```diff
@@ -1040,6 +1054,246 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
+                castNode.AddAttribute("to", t);
+                node = ctx.CreateNode(opType, castOutput, labelEncoderOutput, ctx.GetNodeName(opType));
+                var values = Array.ConvertAll(_valueMap.GetKeys<T>(), item => Convert.ToInt64(item));
+                node.AddAttribute("keys_int64s", values);
+            }
+
+            private void CastInputToFloat<T>(OnnxContext ctx, out OnnxNode node, string srcVariableName, string opType, string labelEncoderOutput)
+            {
+                var srcShape = ctx.RetrieveShapeOrNull(srcVariableName);
+                var castOutput = ctx.AddIntermediateVariable(new VectorDataViewType(NumberDataViewType.Single, (int)srcShape[1]), "castOutput");
+                var castNode = ctx.CreateNode("Cast", srcVariableName, castOutput, ctx.GetNodeName("Cast"), "");
+                var t = InternalDataKindExtensions.ToInternalDataKind(DataKind.Single).ToType();
+                castNode.AddAttribute("to", t);
+                node = ctx.CreateNode(opType, castOutput, labelEncoderOutput, ctx.GetNodeName(opType));
+                var values = Array.ConvertAll(_valueMap.GetKeys<T>(), item => Convert.ToSingle(item));
+                node.AddAttribute("keys_floats", values);
+            }
+
+            private bool SaveAsOnnxCore(OnnxContext ctx, string srcVariableName, string dstVariableName)
+            {
+                const int minimumOpSetVersion = 9;
+                ctx.CheckOpSetVersion(minimumOpSetVersion, LoaderSignature);
+                OnnxNode node;
+                string opType = "LabelEncoder";
+                var labelEncoderInput = srcVariableName;
+                var srcShape = ctx.RetrieveShapeOrNull(srcVariableName);
+                var typeValue = _valueMap.ValueColumn.Type;
+                var typeKey = _valueMap.KeyColumn.Type;
+
+                var labelEncoderOutput = (typeValue == NumberDataViewType.Single || typeValue == TextDataViewType.Instance || typeValue == NumberDataViewType.Int64) ? dstVariableName :
+                    (typeValue == NumberDataViewType.Double || typeValue == BooleanDataViewType.Instance) ? ctx.AddIntermediateVariable(new VectorDataViewType(NumberDataViewType.Single, (int)srcShape[1]), "LabelEncoderOutput") :
+                    ctx.AddIntermediateVariable(new VectorDataViewType(NumberDataViewType.Int64, (int) srcShape[1]), "LabelEncoderOutput");
+
+                // The LabelEncoder operator doesn't support mappings between the same type and only supports mappings between int64s, floats, and strings.
+                // As a result, we need to cast most inputs and outputs. In order to avoid as many unsupported mappings, we cast keys that are of NumberDataTypeView
+                // to strings and values of NumberDataTypeView to int64s.
+                // String -> String mappings can't be supported.
+                if (typeKey == NumberDataViewType.Int64)
```

Comment:
I'm left wondering if it's possible re refactor this big if-else statement into something that's reusable, but as discussed offline it might not be possible as each case handles things in a particular way and you might end up re-writing the if-else blocks anyway, and not saving much lines of code.